tests/ui/modulo_arithmetic.rs

```diff
@@ -0,0 +1,32 @@
+#![warn(clippy::modulo_arithmetic)]
+#![allow(
+    unused,
+    clippy::shadow_reuse,
+    clippy::shadow_unrelated,
+    clippy::no_effect,
+    clippy::unnecessary_operation
+)]
+
+fn main() {
+    // Lint on signed integral and floating point numbers
+    -1 % 2;
+    1 % -2;
+    1 % 2;
+    -1 % -2;
+
+    -1.6 % 2.1;
+    1.6 % -2.1;
+    1.6 % 2.1;
+    -1.6 % -2.1;
+
```

Comment:
How about testing `f64`s and `i128`s? Also suffixed literals (e.g. `1i16 % 42`)?

Reply:
Just added a bunch of different kinds of tests... Maybe even a little bit too much :)

Reply:
Oh yes, I think there are too many...

```
Error: stderr files exceeding limit of 275 lines:
../tests/ui/modulo_arithmetic.stderr
```